### PR TITLE
Mainstems method for FIM 3

### PR DESCRIPTION
**Pull request by BrianAvant-NOAA**
Adding mainstems method to FIM 3 Issue #28 

## Additions

- fim_run.sh
     * added extent argument to support mainstem method; options are MS or FR. This new argument is required.
     * example call:
```fim_run.sh -u /data/temp/parameter_adj_files/dev_fim_ble.lst -c /foss_fim/config/params_template.env -n dev_faf2237_huc8 -e MS -j 2```

- run_by_unit.sh
     * added catches in the MS workflow to handle HUCs with no AHPs forecasting point or incoming streams and exit run_by_unit.sh gracefully.
     * updated arguments to snap_and_clip_to_nhd.py and add_crosswalk.py
     * assigns raster variables to new generic name to handle both FR and MS extents

- snap_and_clip_to_nhd.py 
     * FR - no change; uses NWM headwater points to subset NHDPlus HR stream network
     * MS - uses AHPs sites as well as incoming streams to subset NHDPlus HR stream network instead of NWM headwater points

- fr_to_ms_raster_mask.py 
After split_flows.py, slope, rem, and fdr grids are masked to buffered MS  stream network. 
     * The default buffer distance is 7000 m, based on FIM 2.3.3. 
     * Buffer parameter is added to the environment file so it can be adjusted easily for calibration and sensitivity analysis.

- add_crosswalk.py now uses two crosswalk methods for FR and MS. 
     * FR still uses method where zonal statistics are calculated on the NWM and FIM catchment rasters to find the NWM catchment with the majority overlap with a FIM catchment. 
     * MS method uses a different technique because the FIM catchments are much larger and selection of a NWM tributary would result in a low flow on the mainstem. The crosswalk for mainstem uses the midpoint of each mainstem stream segment to spatially join attributes with NWM catchments. A similar technique was used for FIM 2.3.3.

## Changes

- cleaned up comments in split_flows.py
- cleaned up comments in buildstreamtraversal.py
- checks for empty streams layer in filter_catchments_and_add_attributes.py; exits gracefully if empty
- majority.geojson is depreciated; rasterstats zonal statistics is now called inside of add_crosswalk.py instead of stand alone call in run_by_unit.sh
- updated mannings_calibration_run.sh and mannings_run_by_set.sh to handle new add_crosswalk.py arguments

## Testing
FIM 3 MS compared to FIM 2.3.3 MS
- CSI - 100yr
image

- CSI - 500yr
image

- CSI - aggregate score across 48 HUC8s
image

There are two main reasons why the MS FIM 3 scoring performs worse than FIM 2.3.3. relative to FR:
1. Missing Flows
In the current evaluation configuration, flows are only available at NWM 'feature_id's where there is an intersection with a BLEs cross section. Depending on the density of the cross sections for a given area this can create gaps in coverage that are not accounted for in the FIM crosswalk. This issue is more prevalent in the mainstem feature branch due to the large FIM catchemnts but it is likely causing some errors in the FR scoring as well. Below is an example (HUC 11140103: -2.4%/100yr,  0.76%/500yr):

image
These two catchments have stage-discharge relationships defined in the hydroTable.csv and are crosswalked to a NWM stream that outputs flow but no inundation is produced because there was no intersection with a BLE cross section at that NWM segment.

2. Catchment boundaries based on poor hydroconditioning
In some of the worst performing HUCs, the catchment boundary restricts inundation even after removing tributaries. In the example below (HUC 11140104: -26.08%/100yr,  -23.33%/500yr), FIM 3 is unable to predict inundation of the right side of the channel:
image

This is because the catchment boundary acts as a wall preventing inundation of adjacent pixels:
image

Refining the hydroconditioning methods to produce better catchment representation should result in better inundation predictions. Here is an example of FR catchment boundaries of the same area using the AGREE DEM hydroconditioning method (PR #84):
image

The MS approach should increase the catchment boundaries even more, allowing for more inundation along the right side of the floodplain.

## Todos

- currently incoming streams intersecting HUC boundaries are assumed to be mainstems but there are several cases where this isn't true. This method would benefit from the resolution of issue #58.

### CODE REVIEW

**Comment by BrianAvant-NOAA** [comment]
metrics after merged with dev_4768370
100yr
image

500yr 
image

**Comment by BrianAvant-NOAA** [comment]
Minor update to crosswalk to enforce preservation of leading zeroes in HUC8 and HydroID columns.

File: lib/run_by_unit.sh
Comment (TrevorGrout-NOAA) [inline lib/run_by_unit.sh:239]: It seems that convention is to add "_MS". Please consider changing slopes_d8_dem_metersMS to slopes_d8_dem_meters_MS (here and other places if necessary) or specify reasoning for keeping.

**Comment by TrevorGrout-NOAA** [comment]
This feature branch enables a new capability to run Mainstems configuration.  I approve this pull request once all minor comments have been addressed. Tickets for enhancements have been submitted and they include:
1) Creation of a national mainstems network (#101)
2) Investigation of mainstems headwater mapping extending much further upstream than FIM 3 (#99)
3) Disregard rating curves from very small catchments and use neighboring rating curves instead (#100)
4) Comparison of crosswalk methods from NHD to NWM (#98)
5) Implement a consistent HydroID between MS and FR (#97)

Feature Branch: dev-alt-mainstems
Pull Request: Mainstems method for FIM 3 #89
Author: Brian Avant
Reviewer: Trevor Grout
 
_Developer responsible for feature enhancement will submit documentation that provides the following:
A brief description of the feature enhancement (use images as needed)_
**The pull request adequately describes the feature enhancement as well potential reasons for scores. Code was reviewed with Brian and looks ok.**
_Skill and computational metrics that demonstrate impact of feature enhancement when compared to skill and computation efficiency of dev branch.
Document test cases used._
**Brian provided the command he used to run his feature branch. This command was verified. Test cases were performed for all hucs in dev_fim_ble.lst.**
_Explanation of differences in skill and computational metrics (regression or otherwise)._
**The differences between FIM 2 and FIM 3 were documented in detail including potential reasons for the regressions.  Of note, Item 2 indicating basin boundary issues appears to be resolved with the addition of the AGREE methodology.**  
image
_If enhancement leads to regression, justification for merging regression into dev branch._
**Due to the assumption that streams which cross the huc boundary are considered mainstems, the comparison between FIM 2 and FIM 3 isn’t really a fair comparison. In some HUCs the FIM 2 is much different stream/mapping than FIM 3. Ticket #101 should be a permanent fix. The current assumption of incoming streams being treated as mainstems is appropriate so that mainstem rivers that flow from HUC to HUC are accurately captured.** 
image
image
image
_**Reviewers will provide initial review within 3 working days of documenting intention to review. Reviewer will:**_
_Execute code and document successful execution._
**The dev-alt-mainstems feature branch was checked out and the code was successfully run for all hucs in the /data/temp/parameter_adj_files/dev_fim_ble.lst**
_Check relevant outputs for fidelity._
**Changes listed in PR were tested and outputs were verified.**
_Run at least one test case provided in Pull Request and confirm reported values._
**Noted error with autoeval for 08040101. It appears the issue is the leading zero is truncated in the Hydrotable.csv.  Brian has pushed fix to feature branch. After this fix, the feature branch was run for both full res and mainstems so that no regressions were evident from FR as well ability to replicate MS scores.** 

**Full res comparisons to the latest dev (47683) was compared to this feature branch and all CSI values were identical except for 6 HUCs (100 & 500) which were all less than 0.2% different. It is likely that these HUCs are affected by known issue #80. All other HUCs were identical. Satisfied that this PR does not adversely affect current FR configuration.  It was noted that with the FR configuration, no datasets with _MS were produced (as expected).  Output REM files for FR covers appropriate extent and MS also covers an appropriate extent.** 

total_area | dev_4768370 | new_feature | eval | site | branch | Diff
-- | -- | -- | -- | -- | -- | --
CSI | 0.619330912 | 0.619331 | 100yr | 11090202 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.646060421 | 0.64606 | 500yr | 11090202 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.471458126 | 0.471458 | 100yr | 11090203 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.502789773 | 0.50279 | 500yr | 11090203 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.568073446 | 0.568073 | 100yr | 11090204 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.582176473 | 0.582176 | 500yr | 11090204 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.581478136 | 0.581478 | 100yr | 11100303 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.610252414 | 0.610252 | 500yr | 11100303 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.595266017 | 0.595266 | 100yr | 11130302 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.634596203 | 0.634596 | 500yr | 11130302 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.564258486 | 0.564258 | 100yr | 11130303 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.610474664 | 0.610475 | 500yr | 11130303 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.584995784 | 0.584996 | 100yr | 11130304 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.601005228 | 0.601005 | 500yr | 11130304 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.613207507 | 0.613208 | 100yr | 11130301 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.650990323 | 0.65099 | 500yr | 11130301 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.596849095 | 0.596849 | 100yr | 12020001 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.620396552 | 0.620397 | 500yr | 12020001 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.508959225 | 0.509397 | 100yr | 12020002 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0.000438
CSI | 0.535930407 | 0.536414 | 500yr | 12020002 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0.000483
CSI | 0.549833678 | 0.549834 | 100yr | 12020003 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.570410937 | 0.570411 | 500yr | 12020003 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.656207889 | 0.656208 | 100yr | 12020004 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.678860544 | 0.678861 | 500yr | 12020004 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.585362711 | 0.585363 | 100yr | 12020005 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.612015525 | 0.612016 | 500yr | 12020005 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.53740296 | 0.537403 | 100yr | 12020006 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.556805705 | 0.556806 | 500yr | 12020006 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.394352355 | 0.394352 | 100yr | 12020007 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.40550255 | 0.405503 | 500yr | 12020007 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.545414257 | 0.545414 | 100yr | 12030104 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.572289826 | 0.57229 | 500yr | 12030104 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.560171685 | 0.560172 | 100yr | 12030107 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.594610084 | 0.59461 | 500yr | 12030107 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.518040299 | 0.51804 | 100yr | 12030108 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.561239748 | 0.56124 | 500yr | 12030108 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.504913451 | 0.504913 | 100yr | 12030109 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.535662368 | 0.535662 | 500yr | 12030109 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.611713935 | 0.611714 | 100yr | 12040101 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.634416169 | 0.634416 | 500yr | 12040101 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.563545438 | 0.563545 | 100yr | 12040103 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.597571222 | 0.597571 | 500yr | 12040103 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.493882528 | 0.493883 | 100yr | 12060202 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.527905211 | 0.527905 | 500yr | 12060202 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.577991985 | 0.577992 | 100yr | 12090301 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.617280594 | 0.617281 | 500yr | 12090301 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.577312392 | 0.577312 | 100yr | 12100201 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.627146119 | 0.627146 | 500yr | 12100201 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.639475814 | 0.639476 | 100yr | 12100202 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.669694168 | 0.669694 | 500yr | 12100202 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.597747096 | 0.597747 | 100yr | 12100203 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.634342026 | 0.634342 | 500yr | 12100203 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.346776023 | 0.345053 | 100yr | 13020101 | tsg_PR_dev_alt_mainstems_huc8_final_fr | -0.00172
CSI | 0.365321578 | 0.36376 | 500yr | 13020101 | tsg_PR_dev_alt_mainstems_huc8_final_fr | -0.00156
CSI | 0.350317962 | 0.350318 | 100yr | 13020102 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.372064803 | 0.372065 | 500yr | 13020102 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.62290812 | 0.622908 | 100yr | 11010004 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.645624473 | 0.645624 | 500yr | 11010004 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.78200313 | 0.782003 | 100yr | 11010009 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.811931851 | 0.811932 | 500yr | 11010009 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.538736195 | 0.538736 | 100yr | 11010014 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.533726305 | 0.533726 | 500yr | 11010014 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.57523686 | 0.575048 | 100yr | 11110202 | tsg_PR_dev_alt_mainstems_huc8_final_fr | -0.00019
CSI | 0.592023551 | 0.591856 | 500yr | 11110202 | tsg_PR_dev_alt_mainstems_huc8_final_fr | -0.00017
CSI | 0.581550966 | 0.581551 | 100yr | 11110205 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.584703928 | 0.584704 | 500yr | 11110205 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.630814217 | 0.630814 | 100yr | 11050003 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.65054269 | 0.650543 | 500yr | 11050003 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.509988601 | 0.509989 | 100yr | 11120303 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.530562172 | 0.530562 | 500yr | 11120303 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.497878185 | 0.497878 | 100yr | 11130202 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.541742546 | 0.541743 | 500yr | 11130202 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.50371734 | 0.503717 | 100yr | 11130203 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.541695663 | 0.541696 | 500yr | 11130203 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.56523744 | 0.565237 | 100yr | 11130208 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.614410028 | 0.61441 | 500yr | 11130208 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.634955286 | 0.634955 | 100yr | 11140102 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.656209914 | 0.65621 | 500yr | 11140102 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.565505047 | 0.565505 | 100yr | 11140103 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.578552604 | 0.578553 | 500yr | 11140103 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.656933147 | 0.656999 | 100yr | 11140104 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 6.59E-05
CSI | 0.695102188 | 0.695215 | 500yr | 11140104 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0.000113
CSI | 0.618940029 | 0.61894 | 100yr | 11110104 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.640575602 | 0.640576 | 500yr | 11110104 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.53879376 | 0.538794 | 100yr | 11140105 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.549146556 | 0.549147 | 500yr | 11140105 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.6314864 | 0.631486 | 100yr | 11070103 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.628333339 | 0.628333 | 500yr | 11070103 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.534778109 | 0.534778 | 100yr | 8040101 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.558191509 | 0.558192 | 500yr | 8040101 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.642344098 | 0.642344 | 100yr | 8040203 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.646121035 | 0.646121 | 500yr | 8040203 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.399131068 | 0.397621 | 100yr | 12060102 | tsg_PR_dev_alt_mainstems_huc8_final_fr | -0.00151
CSI | 0.461716127 | 0.46039 | 500yr | 12060102 | tsg_PR_dev_alt_mainstems_huc8_final_fr | -0.00133
CSI | 0.557973536 | 0.557974 | 100yr | 8070201 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.557104676 | 0.557105 | 500yr | 8070201 | tsg_PR_dev_alt_mainstems_huc8_final_fr | 0
CSI | 0.49123921 | 0.489935 | 100yr | 12110110 | tsg_PR_dev_alt_mainstems_huc8_final_fr | -0.0013
CSI | 0.537220858 | 0.536561 | 500yr | 12110110 | tsg_PR_dev_alt_mainstems_huc8_final_fr | -0.00066


**Upon comparison, mainstem metrics were identical for almost all HUCs. 3 exceptions occurred, 12020002 and 12060102 which both exhibited discrepancies (#80) and 13020101 which failed for some reason on run_test_case. Able to replicate scores adequately. See Table Below**


total_area | new_feature | eval | site | branch | TSG Run | Diff
-- | -- | -- | -- | -- | -- | --
CSI | 0.551596 | 100yr | 11090202 | dev_4768370_huc8_MS | 0.551596 | 0
CSI | 0.349563 | 100yr | 11090203 | dev_4768370_huc8_MS | 0.349563 | 0
CSI | 0.472502 | 100yr | 11090204 | dev_4768370_huc8_MS | 0.472502 | 0
CSI | 0.447352 | 100yr | 11100303 | dev_4768370_huc8_MS | 0.447352 | 0
CSI | 0.467985 | 100yr | 11130302 | dev_4768370_huc8_MS | 0.467985 | 0
CSI | 0.363007 | 100yr | 11130303 | dev_4768370_huc8_MS | 0.363007 | 0
CSI | 0.435904 | 100yr | 11130304 | dev_4768370_huc8_MS | 0.435904 | 0
CSI | 0.288133 | 100yr | 11130301 | dev_4768370_huc8_MS | 0.288133 | 0
CSI | 0.291911 | 100yr | 12020001 | dev_4768370_huc8_MS | 0.291911 | 0
CSI | 0.315676 | 100yr | 12020002 | dev_4768370_huc8_MS | 0.315319 | -0.00036
CSI | 0.354592 | 100yr | 12020003 | dev_4768370_huc8_MS | 0.354592 | 0
CSI | 0.450181 | 100yr | 12020004 | dev_4768370_huc8_MS | 0.450181 | 0
CSI | 0.220039 | 100yr | 12020005 | dev_4768370_huc8_MS | 0.220039 | 0
CSI | 0.32297 | 100yr | 12020006 | dev_4768370_huc8_MS | 0.32297 | 0
CSI | 0.233333 | 100yr | 12020007 | dev_4768370_huc8_MS | 0.233333 | 0
CSI | 0.343377 | 100yr | 12030104 | dev_4768370_huc8_MS | 0.343377 | 0
CSI | 0.278354 | 100yr | 12030107 | dev_4768370_huc8_MS | 0.278354 | 0
CSI | 0.27862 | 100yr | 12030108 | dev_4768370_huc8_MS | 0.27862 | 0
CSI | 0.214134 | 100yr | 12030109 | dev_4768370_huc8_MS | 0.214134 | 0
CSI | 0.522448 | 100yr | 12040101 | dev_4768370_huc8_MS | 0.522448 | 0
CSI | 0.361824 | 100yr | 12040103 | dev_4768370_huc8_MS | 0.361824 | 0
CSI | 0.338391 | 100yr | 12060202 | dev_4768370_huc8_MS | 0.338391 | 0
CSI | 0.395964 | 100yr | 12090301 | dev_4768370_huc8_MS | 0.395964 | 0
CSI | 0.413121 | 100yr | 12100201 | dev_4768370_huc8_MS | 0.413121 | 0
CSI | 0.489558 | 100yr | 12100202 | dev_4768370_huc8_MS | 0.489558 | 0
CSI | 0.459309 | 100yr | 12100203 | dev_4768370_huc8_MS | 0.459309 | 0
CSI | 0.282899 | 100yr | 13020101 | dev_4768370_huc8_MS | #N/A | #N/A
CSI | 0.280337 | 100yr | 13020102 | dev_4768370_huc8_MS | 0.280337 | 0
CSI | 0.583071 | 100yr | 11010004 | dev_4768370_huc8_MS | 0.583071 | 0
CSI | 0.556185 | 100yr | 11010009 | dev_4768370_huc8_MS | 0.556185 | 0
CSI | 0.251886 | 100yr | 11010014 | dev_4768370_huc8_MS | 0.251886 | 0
CSI | 0.531213 | 100yr | 11110202 | dev_4768370_huc8_MS | 0.531213 | 0
CSI | 0.437007 | 100yr | 11050003 | dev_4768370_huc8_MS | 0.437007 | 0
CSI | 0.365434 | 100yr | 11120303 | dev_4768370_huc8_MS | 0.365434 | 0
CSI | 0.369927 | 100yr | 11130202 | dev_4768370_huc8_MS | 0.369927 | 0
CSI | 0.425967 | 100yr | 11130203 | dev_4768370_huc8_MS | 0.425967 | 0
CSI | 0.311259 | 100yr | 11130208 | dev_4768370_huc8_MS | 0.311259 | 0
CSI | 0.495458 | 100yr | 11140102 | dev_4768370_huc8_MS | 0.495458 | 0
CSI | 0.532842 | 100yr | 11140103 | dev_4768370_huc8_MS | 0.532842 | 0
CSI | 0.325738 | 100yr | 11140104 | dev_4768370_huc8_MS | 0.325738 | 0
CSI | 0.475556 | 100yr | 11110104 | dev_4768370_huc8_MS | 0.475556 | 0
CSI | 0.357306 | 100yr | 11140105 | dev_4768370_huc8_MS | 0.357306 | 0
CSI | 0.525317 | 100yr | 11070103 | dev_4768370_huc8_MS | 0.525317 | 0
CSI | 0.27164 | 100yr | 12060102 | dev_4768370_huc8_MS | 0.272164 | 0.000524



total_area | new_feature | eval | site | branch | TSG_Run | Diff
-- | -- | -- | -- | -- | -- | --
CSI | 0.572845 | 500yr | 11090202 | dev_4768370_huc8_MS | 0.572845 | 0
CSI | 0.347336 | 500yr | 11090203 | dev_4768370_huc8_MS | 0.347336 | 0
CSI | 0.487741 | 500yr | 11090204 | dev_4768370_huc8_MS | 0.487741 | 0
CSI | 0.467976 | 500yr | 11100303 | dev_4768370_huc8_MS | 0.467976 | 0
CSI | 0.478662 | 500yr | 11130302 | dev_4768370_huc8_MS | 0.478662 | 0
CSI | 0.401213 | 500yr | 11130303 | dev_4768370_huc8_MS | 0.401213 | 0
CSI | 0.446546 | 500yr | 11130304 | dev_4768370_huc8_MS | 0.446546 | 0
CSI | 0.300895 | 500yr | 11130301 | dev_4768370_huc8_MS | 0.300895 | 0
CSI | 0.309895 | 500yr | 12020001 | dev_4768370_huc8_MS | 0.309895 | 0
CSI | 0.335998 | 500yr | 12020002 | dev_4768370_huc8_MS | 0.335675 | -0.00032
CSI | 0.376111 | 500yr | 12020003 | dev_4768370_huc8_MS | 0.376111 | 0
CSI | 0.469949 | 500yr | 12020004 | dev_4768370_huc8_MS | 0.469949 | 0
CSI | 0.23815 | 500yr | 12020005 | dev_4768370_huc8_MS | 0.23815 | 0
CSI | 0.37499 | 500yr | 12020006 | dev_4768370_huc8_MS | 0.37499 | 0
CSI | 0.236736 | 500yr | 12020007 | dev_4768370_huc8_MS | 0.236736 | 0
CSI | 0.348051 | 500yr | 12030104 | dev_4768370_huc8_MS | 0.348051 | 0
CSI | 0.298321 | 500yr | 12030107 | dev_4768370_huc8_MS | 0.298321 | 0
CSI | 0.312931 | 500yr | 12030108 | dev_4768370_huc8_MS | 0.312931 | 0
CSI | 0.221151 | 500yr | 12030109 | dev_4768370_huc8_MS | 0.221151 | 0
CSI | 0.551022 | 500yr | 12040101 | dev_4768370_huc8_MS | 0.551022 | 0
CSI | 0.40025 | 500yr | 12040103 | dev_4768370_huc8_MS | 0.40025 | 0
CSI | 0.374304 | 500yr | 12060202 | dev_4768370_huc8_MS | 0.374304 | 0
CSI | 0.423923 | 500yr | 12090301 | dev_4768370_huc8_MS | 0.423923 | 0
CSI | 0.459395 | 500yr | 12100201 | dev_4768370_huc8_MS | 0.459395 | 0
CSI | 0.507977 | 500yr | 12100202 | dev_4768370_huc8_MS | 0.507977 | 0
CSI | 0.485114 | 500yr | 12100203 | dev_4768370_huc8_MS | 0.485114 | 0
CSI | 0.274682 | 500yr | 13020101 | dev_4768370_huc8_MS | #N/A | #N/A
CSI | 0.282648 | 500yr | 13020102 | dev_4768370_huc8_MS | 0.282648 | 0
CSI | 0.62534 | 500yr | 11010004 | dev_4768370_huc8_MS | 0.62534 | 0
CSI | 0.588652 | 500yr | 11010009 | dev_4768370_huc8_MS | 0.588652 | 0
CSI | 0.225061 | 500yr | 11010014 | dev_4768370_huc8_MS | 0.225061 | 0
CSI | 0.539008 | 500yr | 11110202 | dev_4768370_huc8_MS | 0.539008 | 0
CSI | 0.39663 | 500yr | 11050003 | dev_4768370_huc8_MS | 0.39663 | 0
CSI | 0.369815 | 500yr | 11120303 | dev_4768370_huc8_MS | 0.369815 | 0
CSI | 0.442515 | 500yr | 11130202 | dev_4768370_huc8_MS | 0.442515 | 0
CSI | 0.471909 | 500yr | 11130203 | dev_4768370_huc8_MS | 0.471909 | 0
CSI | 0.359996 | 500yr | 11130208 | dev_4768370_huc8_MS | 0.359996 | 0
CSI | 0.520592 | 500yr | 11140102 | dev_4768370_huc8_MS | 0.520592 | 0
CSI | 0.560466 | 500yr | 11140103 | dev_4768370_huc8_MS | 0.560466 | 0
CSI | 0.348476 | 500yr | 11140104 | dev_4768370_huc8_MS | 0.348476 | 0
CSI | 0.536537 | 500yr | 11110104 | dev_4768370_huc8_MS | 0.536537 | 0
CSI | 0.36433 | 500yr | 11140105 | dev_4768370_huc8_MS | 0.36433 | 0
CSI | 0.489959 | 500yr | 11070103 | dev_4768370_huc8_MS | 0.489959 | 0
CSI | 0.31722 | 500yr | 12060102 | dev_4768370_huc8_MS | 0.317471 | 0.000251

**Comment by TrevorGrout-NOAA** [comment]
@nickchadwick-noaa, @frsalas-noaa, @BrianAvant-NOAA  I approve this pull request.

**Review by TrevorGrout-NOAA — APPROVED** [review]

**Pull request merged**
